.azure-pipelines/bazel.yml

```diff
@@ -52,7 +52,10 @@ steps:
       ENVOY_RBE: "1"
       BAZEL_BUILD_EXTRA_OPTIONS: "--config=remote-ci --jobs=$(RbeJobs) ${{ parameters.bazelBuildExtraOptions }}"
       BAZEL_REMOTE_CACHE: grpcs://remotebuildexecution.googleapis.com
-      BAZEL_REMOTE_INSTANCE: projects/envoy-ci/instances/default_instance
+      ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
```

Comment:
This logic needs to modify the cache-silo-key instead of the RBE instance name. It will achieve the same effect. Basically something like so:
```
  ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
    BAZEL_BUILD_EXTRA_OPTIONS: "--config=remote-ci --jobs=$(RbeJobs) --remote_default_exec_properties=cache-silo-key=\"cache-silo-$(System.PullRequest.TargetBranch)\" ${{ parameters.bazelBuildExtraOptions }}"
  ${{ if ne(variables['Build.Reason'], 'PullRequest') }}:
    BAZEL_BUILD_EXTRA_OPTIONS: "--config=remote-ci --jobs=$(RbeJobs) --remote_default_exec_properties=\"cache-silo-key=cache-silo-$(System.PullRequest.SourceBranch)\" ${{ parameters.bazelBuildExtraOptions }}"
```

Reply:
cool we need to make it work for both local and RBE but i can give that a try